server/views/formBuilder.njk

```diff
@@ -2,178 +2,219 @@
-                    value: option.value,
-                    checked: formResponses | formResponsesContainsValue(input.name, option.value) or additionalCondition.data | additionalConditionDataContainsValue(input.name, option.value),
-                    conditional: {
-                        html: formBuilder(option.conditional, additionalCondition, validationErrors, formResponses)
-                    } if option.conditional
-                }), options) %}
-            {% endfor %}
-
-            {{ govukRadios({
-                idPrefix: input.name,
-                name: input.name,
-                fieldset: {
-                    legend: {
-                        text: input.label,
-                        classes: "govuk-!-font-weight-bold"
-                    }
-                },
-                items: options,
-                errorMessage: validationErrors | findError(input.name)
-            }) }}
-        {% endif %}
-
-        {% if input.type == 'datePicker' %}
-            {{ datePicker({
-                id: input.name,
-                label: {
-                    text: input.label,
-                    classes: "govuk-!-font-weight-bold"
-                },
-                hint: {
-                    text: "For example, 12 11 2022"
-                },
-                errorMessage: validationErrors | findError(input.name),
-                formResponses: additionalCondition.data | getAdditionalConditionSimpleDateValue(input.name) | fillFormResponse(formResponses)
-            }) }}
-        {% endif %}
-
-        {% if input.type == 'timePicker' %}
-            {{ timePicker({
-                id: input.name,
-                label: {
-                    text: input.label,
-                    classes: "govuk-!-font-weight-bold"
-                },
-                hint: {
-                    text: "For example, 9:30am or 2:55pm"
-                },
-                errorMessage: validationErrors | findError(input.name),
-                formResponses: additionalCondition.data | getAdditionalConditionSimpleTimeValue(input.name) | fillFormResponse(formResponses)
-            }) }}
-        {% endif %}
-
-        {% if input.type == 'address' %}
-            {% call govukFieldset({
-                legend: {
-                    text: "Enter address",
-                    classes: "govuk-!-font-weight-bold"
-                },
-                classes: 'govuk-!-margin-top-9'
-            }) %}
-                {{ govukInput({
-                    id: 'addressLine1',
-                    label: {
-                        html: 'Building and street <span class="govuk-visually-hidden">line 1 of 2</span>'
+        {% set validationError = validationErrors | findError(input.name) %}
+        {% set formValues = getFormResponses(formResponses, input.name) or getValueFromAdditionalCondition(additionalCondition, input.name) %}
+
+        <div data-module="moj-add-another">
+            {% if input.type == 'text' %}
```

Comment:
Will this always create a div for 'moj-add-another' ? For ALL input types?  (No surrounding {% if  %})

Reply:
The `data-module` attribute will enable the add another client side javascript functionality for all data types, but it will not show the button unless addAnother is provided in the field config. 

So yes even if addAnother is not provided in the config, it will be wrapped in a div with the module add another, but it will not have any effect. In the interest of keeping this already complicated form builder as simple as possible, I went with wrapping all inputs in this